hudi-client/hudi-flink-client/src/main/java/org/apache/hudi/client/common/HoodieFlinkEngineContext.java

```diff
@@ -218,6 +222,12 @@ public ReaderContextFactory<?> getReaderContextFactory(HoodieTableMetaClient met
     return (ReaderContextFactory<?>) ReflectionUtils.loadClass("org.apache.hudi.table.format.FlinkReaderContextFactory", metaClient);
   }
 
+  public void dropIndex(HoodieConfig config, List<String> metadataPartitions) {
+    try (BaseHoodieWriteClient client = new HoodieFlinkWriteClient(this, (HoodieWriteConfig) config)) {
```

Comment:
1. the `#dropIndex` does not belong here;
2. use `SupportsUpgradeDowngrade#getWriteClient` to fetch the write client;
3. be caution to the write config we used in 2 while constructing the write client to avoid starting the embedded timeline servier.